src/DOM/WebStorage/Internal/Generic.purs

```diff
@@ -0,0 +1,28 @@
+module DOM.WebStorage.Internal.Generic where
+
+import Prelude
+
+import Data.Argonaut.Decode (class DecodeJson, gDecodeJson)
+import Data.Argonaut.Encode (class EncodeJson, gEncodeJson)
+import Data.Generic (class Generic, gCompare, gEq, gShow)
+
+newtype TranscodeG a = TranscodeG a
+derive instance genericGTransocde :: Generic a => Generic (TranscodeG a)
+
+instance showTranscodeG :: Generic a => Show (TranscodeG a) where
+  show = gShow
+
+instance eqTranscodeG ::Generic a => Eq (TranscodeG a) where
+  eq = gEq
+
+instance ordTranscodeG :: Generic a => Ord (TranscodeG a) where
+  compare = gCompare
+
+instance encodeTranscodeG :: Generic a => EncodeJson (TranscodeG a) where
+  encodeJson = gEncodeJson
+
+instance decodeTranscodeG :: Generic a => DecodeJson (TranscodeG a) where
+  decodeJson = gDecodeJson
+
+unTranscodeG :: forall a. TranscodeG a -> a
+unTranscodeG (TranscodeG item) = item
```

Comment:
Neat trick! But I think I'd like a separate type class for generic encoding better - this would make user code as clear as before again. (No TranscodeG distraction.) In other words, in addition to JSONStorage a GenericStorage class in a file Generic.purs, with the same functions, but Generic constraint instead of the ToJSON/FromJSON constraints.